[Runtime Fields] Replace enzyme with RTL

kapral18 opened the pull request:
Closes https://github.com/elastic/kibana/issues/241532

## Summary
- Migrate Runtime Fields UI tests from testbed/Enzyme to React Testing Library.
- Remove legacy fake-timer patterns and make async waits explicit.
- Add lightweight mocks for CodeEditor/ComboBox to keep unit tests deterministic.

## Test titles table of change
To make it easier to verify individual tests here is a table of test titles before and after change
for each changed file in the PR. 
- https://gist.github.com/kapral18/709502e63dafb2b798e6762ac303f775

## Test plan
- [x] node scripts/jest x-pack/platform/plugins/private/runtime_fields --config x-pack/platform/plugins/private/runtime_fields/jest.config.js
- [x] node scripts/type_check --project x-pack/platform/plugins/private/runtime_fields/tsconfig.json
- [x] node scripts/eslint x-pack/platform/plugins/private/runtime_fields/public

kapral18 committed:
chore(runtime-fields): migrate runtime field tests to RTL

- Replace testbed/Enzyme usage with React Testing Library
- Remove legacy fake timers patterns and stabilize async boundaries
- Add lightweight CodeEditor/EuiComboBox mocks for deterministic unit tests

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/377813)
* Commit: 4eefe576176deb31715f5b896d5a049a715a1bb4

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
<details><summary>Unknown metric groups</summary>

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/707428db222eed5bb7ef882002781f1f0f35b353) | [after](https://github.com/elastic/kibana/commit/4eefe576176deb31715f5b896d5a049a715a1bb4) | diff |
| --- | --- | --- | --- |
| `runtimeFields` | 1 | 0 | -1 |


</details>


### History
* :green_heart: [Build #377521](https://buildkite.com/elastic/kibana-pull-request/builds/377521) succeeded f37d0e8bb2e7bf432f2c7a5e70a4cacc871af7e8
* :yellow_heart: [Build #377390](https://buildkite.com/elastic/kibana-pull-request/builds/377390) was flaky c1a6b1495539b5ebecdd061143df0614ab93defe
* :broken_heart: [Build #377389](https://buildkite.com/elastic/kibana-pull-request/builds/377389) failed 39ed76f8ce4f366b7710d9cd3a30e0fd93a439e0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377813","number":377813,"commit":"4eefe576176deb31715f5b896d5a049a715a1bb4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377521","number":377521,"commit":"f37d0e8bb2e7bf432f2c7a5e70a4cacc871af7e8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377390","number":377390,"commit":"c1a6b1495539b5ebecdd061143df0614ab93defe"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377389","number":377389,"commit":"39ed76f8ce4f366b7710d9cd3a30e0fd93a439e0"}],"number":377813}
buildkite-pr-comment-->

ElenaStoeva commented on x-pack/platform/plugins/private/runtime_fields/public/__jest__/setup_environment.tsx:
```diff
@@ -11,15 +11,35 @@ import '@kbn/code-editor-mock/jest_helper';
+          if (isRecord(evt) && isRecord(evt.target) && typeof evt.target.value === 'string') {
+            const value = evt.target.value;
+            props.onChange([{ label: value, value }]);
+          }
```
Yes, some comments might be helpful to explain why we are doing this

kapral18 replied:
I simplified the mock

ElenaStoeva reviewed (approved):
Lgtm with the small nit about explaining the onChange logic of the EuiComboBox mock.

kibanamachine committed:
chore: simplify EuiComboBox mock in setup_environment.tsx

- Remove unnecessary type guards and helper functions
- Simplify `onChange` handler to directly use `event.target.value`
- Add comment explaining the translation of input value to
EuiComboBox-style payload
